src/Foundation/Commerce/code/ModelInitializers/CartModelBuilder.cs

```diff
@@ -27,25 +27,27 @@ namespace Wooli.Foundation.Commerce.ModelInitializers
 
     using Repositories;
 
+    using Services.Catalog;
+
     using Sitecore.Commerce.Engine.Connect.Entities;
     using Sitecore.Commerce.Entities.Carts;
     using Sitecore.Diagnostics;
 
     [Service(typeof(ICartModelBuilder))]
     public class CartModelBuilder : ICartModelBuilder
     {
-        private readonly ICatalogRepository catalogRepository;
+        private readonly ICatalogService catalogService;
 
         private readonly ICurrencyProvider currencyProvider;
 
         private readonly IEntityMapper entityMapper;
 
         public CartModelBuilder(
-            ICatalogRepository catalogRepository,
+            ICatalogService catalogService,
             IEntityMapper entityMapper,
             ICurrencyProvider currencyProvider)
         {
-            this.catalogRepository = catalogRepository;
+            this.catalogService = catalogService;
             this.entityMapper = entityMapper;
             this.currencyProvider = currencyProvider;
```

Comment:
Add Assert

Reply:
This one will be removed after switching to the new checkout.